src/frontend/src/handler/describe.rs

```diff
@@ -35,30 +35,81 @@ pub fn handle_describe(handler_args: HandlerArgs, table_name: ObjectName) -> Res
     let mut binder = Binder::new(&session);
     let relation = binder.bind_relation_by_name(table_name.clone(), None)?;
     // For Source, it doesn't have table catalog so use get source to get column descs.
-    let (columns, indices): (Vec<ColumnDesc>, Vec<Arc<IndexCatalog>>) = {
-        let (catalogs, indices) = match relation {
-            Relation::Source(s) => (s.catalog.columns, vec![]),
-            Relation::BaseTable(t) => (t.table_catalog.columns, t.table_indexes),
-            Relation::SystemTable(t) => (t.sys_table_catalog.columns, vec![]),
+    let (columns, pk_columns, indices): (Vec<ColumnDesc>, Vec<ColumnDesc>, Vec<Arc<IndexCatalog>>) = {
+        let (column_catalogs, pk_column_catalogs, indices) = match relation {
+            Relation::Source(s) => {
+                let pk_column_catalogs = s
+                    .catalog
+                    .pk_col_ids
+                    .iter()
+                    .map(|&column_id| {
+                        s.catalog
+                            .columns
+                            .iter()
+                            .filter(|x| x.column_id() == column_id)
+                            .exactly_one()
+                            .unwrap()
+                            .clone()
+                    })
+                    .collect_vec();
+                (s.catalog.columns, pk_column_catalogs, vec![])
+            }
+            Relation::BaseTable(t) => {
+                let pk_column_catalogs = t
+                    .table_catalog
+                    .pk()
+                    .iter()
+                    .map(|idx| t.table_catalog.columns[idx.index].clone())
+                    .collect_vec();
+                (t.table_catalog.columns, pk_column_catalogs, t.table_indexes)
+            }
+            Relation::SystemTable(t) => {
+                let pk_column_catalogs = t
+                    .sys_table_catalog
+                    .pk
+                    .iter()
+                    .map(|idx| t.sys_table_catalog.columns[*idx].clone())
+                    .collect_vec();
+                (t.sys_table_catalog.columns, pk_column_catalogs, vec![])
+            }
             _ => {
                 return Err(
                     CatalogError::NotFound("table or source", table_name.to_string()).into(),
                 );
             }
         };
         (
-            catalogs
+            column_catalogs
                 .iter()
                 .filter(|c| !c.is_hidden)
                 .map(|c| c.column_desc.clone())
                 .collect(),
+            pk_column_catalogs
+                .iter()
+                // .filter(|c| !c.is_hidden)
```

Comment:
I'm not sure if we should expose hidden keys like row_id to user. 🤔 Prefer to ignore it.

Reply:
I will remove the comment later, but in some cases I think hidden columns can appear at the prefix of the primary key. In those cases, if we don't expose the hidden columns, users might be confused with why primary point select doesn't work.